operator/src/lib.rs

```diff
@@ -661,26 +472,27 @@ impl SparkState {
     /// Available master urls are hashed and stored as label in the worker pod. If the label differs
     /// from the spec, we need to replace (delete and and) the workers in a rolling fashion.
     pub async fn check_worker_master_urls(&self) -> SparkReconcileResult {
-        if let Some(pod_info) = &self.pod_information {
-            let worker_pods = pod_info.get_all_pods(Some(SparkNodeType::Worker));
-            for pod in &worker_pods {
-                if let Some(labels) = &pod.metadata.labels {
-                    if let Some(label_hashed_master_urls) =
-                        labels.get(pod_utils::MASTER_URLS_HASH_LABEL)
-                    {
-                        let current_hashed_master_urls =
-                            pod_utils::get_hashed_master_urls(&self.spec.master);
-                        if label_hashed_master_urls != &current_hashed_master_urls {
-                            debug!(
-                                "Pod [{}] has an outdated '{}' [{}] - required is [{}], deleting it",
-                                Meta::name(pod),
-                                pod_utils::MASTER_URLS_HASH_LABEL,
-                                label_hashed_master_urls,
-                                current_hashed_master_urls,
-                            );
-                            self.context.client.delete(pod).await?;
-                            return Ok(ReconcileFunctionAction::Requeue(Duration::from_secs(10)));
-                        }
+        let worker_pods = filter_pods_for_type(&self.existing_pods, &SparkNodeType::Worker);
```

Comment:
This is another instance were having existing pods in a HashMap might make sense.